### PR TITLE
Browser detection and naming

### PR DESCRIPTION
**Pull request by userhooke**
**References issue**
#147

**Describe the pull request**
Adding detection for Safari and updating the naming convention.

**Tested in**
- Opera Version:70.0.3728.119
- Safari Version 14.0 
- Firefox Version 80.0b8
- Edge Version 84.0.522.61
- Chrome Version 84.0.4147.125

### CODE REVIEW

File: lib/core/src/support/Utils.ts [hunk 1]
```diff
@@ -21,17 +21,21 @@ export class Utils {
...
-    const extensions: boolean = typeof (window as any).InstallTrigger !== 'undefined'
+    const firefox: boolean = typeof (window as any).InstallTrigger !== 'undefined'
       || (!(document as any).documentNode && (window as any).StyleMedia)
 
     const opera: boolean = (!!(window as any).opr && !!(window as any).addons)
       || !!(window as any).opera || navigator.userAgent.indexOf(' OPR/') >= 0
 
-    // Currently we will not return `Browser.OPERA` but fall back to `.CHROME`
-    return extensions ? Browser.EXTENSIONS
+    const safari: boolean = /constructor/i.test((window as any).HTMLElement)
```
Comment (bausano) [inline lib/core/src/support/Utils.ts:30]: It'd be nice to include a link from which this detector was taken. If something breaks the original link might have a fix.

**Review by bausano — APPROVED** [review]

File: lib/core/src/support/Utils.ts (hunk 1 above)
Comment (userhooke) [reply lib/core/src/support/Utils.ts:30]: I borrowed it from the stackoverflow resource from the comment above 🙂 
```
// Methods are borrowed from the following SO resource which is regularly updated:
// https://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browser#9851769
```

**Review by haydnba — APPROVED** [review]
👍 💛 

Tested the updated feature detection on all browser types and verified that correct browser is identified.

I am wondering if we will need to make further updates to the unit tests for the container service before merging this in? Not certain, perhaps this is something that we can do a little later when we update the actual api wrappers...

Thanks Vladimir!

**Review by kouks — APPROVED** [review]
Thanks for the contribution, looks good.

**Pull request closed without merging**